src/redisboard/views.py

```diff
@@ -97,66 +97,75 @@ def _get_key_details(conn, db, key, page):
-        pipe.execute_command('DEBUG', 'OBJECT', key)
-        pipe.ttl(key)
+        results = pipe.execute()
+        size = results.pop(0)
+        keys = sorted(set(results))
 
-    total_memory = 0
-    volatile_memory = 0
-    persistent_memory = 0
-    total_keys = 0
-    volatile_keys = 0
-    persistent_keys = 0
-    results = pipe.execute()
-    for key, details, ttl in zip(keys, results[::2], results[1::2]):
-        if not isinstance(details, dict):
-            details = dict(_fixup_pair(i.split(b':'))
-                           for i in details.split() if b':' in i)
-
-        length = details[b'serializedlength'] + len(key)
-
-        if ttl:
-            persistent_memory += length
-            persistent_keys += 1
-        else:
-            volatile_memory += length
-            volatile_keys += 1
-        total_memory += length
-        total_keys += 1
+        pipe = server.connection.pipeline()
+        for key in keys:
+            pipe.execute_command('DEBUG', 'OBJECT', key)
+            pipe.ttl(key)
 
-    if total_keys:
-        total_memory = (total_memory / total_keys) * size
-    else:
         total_memory = 0
-
-    if persistent_keys:
-        persistent_memory = (persistent_memory / persistent_keys) * size
-    else:
+        volatile_memory = 0
         persistent_memory = 0
+        total_keys = 0
+        volatile_keys = 0
+        persistent_keys = 0
+        results = pipe.execute()
+        for key, details, ttl in zip(keys, results[::2], results[1::2]):
+            if not isinstance(details, dict):
+                details = dict(_fixup_pair(i.split(b':'))
+                               for i in details.split() if b':' in i)
+
+            length = details[b'serializedlength'] + len(key)
+
+            if ttl:
+                persistent_memory += length
+                persistent_keys += 1
+            else:
+                volatile_memory += length
+                volatile_keys += 1
+            total_memory += length
+            total_keys += 1
+
+        if total_keys:
+            total_memory = (total_memory / total_keys) * size
+        else:
+            total_memory = 0
 
-    if volatile_keys:
-        volatile_memory = (volatile_memory / volatile_keys) * size
-    else:
-        volatile_memory = 0
+        if persistent_keys:
+            persistent_memory = (persistent_memory / persistent_keys) * size
+        else:
+            persistent_memory = 0
 
-    return dict(
-        size=size,
-        total_memory=total_memory,
-        volatile_memory=volatile_memory,
-        persistent_memory=persistent_memory,
-    )
+        if volatile_keys:
+            volatile_memory = (volatile_memory / volatile_keys) * size
+        else:
+            volatile_memory = 0
+
+        return dict(
+            size=size,
+            total_memory=total_memory,
+            volatile_memory=volatile_memory,
+            persistent_memory=persistent_memory,
+        )
+    except ResponseError as exc:
```

Comment:
Yes, i try to follow the existing format in _get_key_info() before trying to be clever. The problem lies in DEBUG OBJECT command which AWS ElastiCache Redis doesn't have.